main/src/modules/Shader.scala

```diff
@@ -0,0 +1,93 @@
+package com.eed3si9n.jarjarabrams
+
+import java.nio.file.{ Files, Path, StandardOpenOption }
+import org.pantsbuild.jarjar.{ JJProcessor, _ }
+import org.pantsbuild.jarjar.util.EntryStruct
+
+object Shader {
+  def shadeDirectory(
+      rules: Seq[ShadeRule],
+      dir: Path,
+      mappings: Seq[(Path, String)],
+      verbose: Boolean
+  ): Unit =
+    if (rules.isEmpty) {}
+    else {
+      val shader = bytcodeShader(rules, verbose)
+      for {
+        (path, name) <- mappings
+        if !Files.isDirectory(path)
+        bytes = Files.readAllBytes(path)
+        _ = Files.delete(path)
+        (shadedBytes, shadedName) <- shader(bytes, name)
+        out = dir.resolve(shadedName)
+        _ = Files.createDirectories(out.getParent)
+        _ = Files.write(out, shadedBytes, StandardOpenOption.CREATE)
+      } yield ()
+    }
+
+  def bytcodeShader(
```

Comment:
Fly-by remark: little typo here, `s/byt/byte`

I'm assuming the monkey patch should get moved to jarjar-abrams proper before this PR is finalized.